### PR TITLE
Fix dictionary conversion to Julia, add more tests

### PR DESCRIPTION
**Pull request by ranjanan**
Fix Dict issue

### CODE REVIEW

**Comment by ranjanan** [comment]
This fixes #87

File: src/convert-base.jl [hunk 1]
```diff
@@ -227,7 +227,7 @@ sexp(d::Associative) = sexp(VecSxp,d)
 
 function rcopy{A<:Associative,S<:VectorSxp}(::Type{A}, s::Ptr{S})
     a = A()
-    K,V = eltype(a)
+    K,V = eltype(a).parameters
```
Comment (simonbyrne) [inline src/convert-base.jl]: I don't think we should be using these internal fields. Better to use:
```
K = keytype(a)
V = valtype(a)
```

Other than that, this looks great, thanks!

**Comment by ranjanan** [comment]
@simonbyrne Incorporated the changes you've suggested.

**Comment by simonbyrne** [comment]
Fantastic, thanks.

Actually, would you mind adding a test as well?

**Comment by ranjanan** [comment]
Ah, I've written a bunch of tests. I thought about opening up another pull request for that. Shall I just add them all to this one?

**Comment by simonbyrne** [comment]
Fantastic. Whatever is easier.

**Comment by ranjanan** [comment]
There seems to be a problem with `rcopy(Cint(1))` . It says `Type cannot be constructed`. This seems to work on my Mac on Julia v0.4.1

**Comment by simonbyrne** [comment]
This is on 0.3, which doesn't support calling bitstypes (such as `Cint(1)`). Add `@compat` outside (e.g. `@compat(Cint(1))`.

Perhaps time has come to deprecate 0.3, though that should be a separate PR.

**Comment by ranjanan** [comment]
I'm afraid `@compat(Cint(1))` doesn't work. It still says `Type cannot be constructed`

**Comment by simonbyrne** [comment]
Hmm, try `@compat(Int32(1))`.

**Comment by ranjanan** [comment]
Oh, that works. Interesting to note that `@compat(Cint(1))` doesn't work even though they're just type-aliases.

**Comment by simonbyrne** [comment]
Macro trickery I guess.

File: src/convert-base.jl [hunk 2]
```diff
@@ -71,7 +71,7 @@ sexp(::Type{CharSxp},sym::Symbol) = sexp(CharSxp,string(sym))
 
 rcopy{T<:AbstractString}(::Type{T},s::CharSxpPtr) = convert(T, bytestring(unsafe_vec(s)))
 rcopy(::Type{Symbol},s::CharSxpPtr) = symbol(rcopy(AbstractString,s))
-
+rcopy(::Type{Int64}, s::CharSxpPtr) = parse(Int, rcopy(s))
```
Comment (simonbyrne) [inline src/convert-base.jl]: This should be `rcopy(::Type{Int}, ....`

**Comment by ranjanan** [comment]
Oh I see. How will changing it to `Int` resolve it? Isn't `Int` typealias for `Int64` on Windows too? Should I also write a method for `Int32`s as well in that case?

**Comment by nalimilan** [comment]
When you'll squash the commits before merging, it would be nice to use a more descriptive title than "update convert-base.jl".

**Comment by ranjanan** [comment]
Oh, I haven't squashed before. Would `git rebase -i HEAD~5` squash all of my commits together?

**Comment by nalimilan** [comment]
Yes, if you change the first word of the lines to `squash` (except for the commits you want to keep of course, whose message can be edited by choosing `edit`). Note that you can keep several commits as long as each of them is meaningful and passes the tests.

**Comment by simonbyrne** [comment]
The problem was only on 32-bit Windows (`Int` aliases to `Int32` on 32-bit systems).

You can also use `git rebase -i master` to rebase from where you diverged from `master`.

**Comment by ranjanan** [comment]
I had to merge a particular commit that was done before I last pulled. I hope that's okay? Or should I merge the two above too?

**Comment by nalimilan** [comment]
You should never have to merge branches: rebase them instead, so that all commits make logical sense. Anyway, the last commit doesn't include any changes, so you should be able to squash it just like the others.

**Comment by ranjanan** [comment]
@nalimilan I should be, but I get this error on `git rebase -i HEAD~2` : `error: could not apply 5e2a5ba... Update convert-base.jl`. Not quite sure why, any ideas?

**Comment by simonbyrne** [comment]
Hmm, you seem to have commits on two different branches that have diverged from `origin/master`.

**Comment by ranjanan** [comment]
Not quite sure why that is. I think I must have committed all these fixes without pulling the latest code in the first place. What that be the reason? And since Travis and Appveyor are happy, could we merge it? Or would the squash have to be done mandatorily?

**Comment by nalimilan** [comment]
Doesn't sound great to pollute the commit log with partial commits like this. A radical solution is to do `git diff master > branch.diff` to get a patch file for the whole branch. Then you can do `git checkout -b backup` to create a copy of the branch (just in case), `git reset --hard master` to destroy all of your changes, and `git apply branch.diff` to apply the patch. Then you can commit it as you do usually.

**Comment by simonbyrne** [comment]
I've rebased and pushed it manually. Thanks for sorting this out!

**Comment by ranjanan** [comment]
Alright thanks! That fixes #87 then.

**Pull request closed without merging**